csi/controller.go

```diff
@@ -587,14 +586,17 @@ func (s *OsdCsiServer) CreateVolume(
 		}
 		newVolumeId = createResp.VolumeId
 	} else {
-		clonedMetadata := getClonedPVCMetadata(locator)
+		labels := locator.GetVolumeLabels()
 		if spec.GetFADAPodName() != "" {
-			clonedMetadata[sdk.FADAPodLabelKey] = spec.GetFADAPodName()
+			labels[api.SpecPurePodName] = spec.GetFADAPodName()
 		}
+		delete(labels, intreePvcNameKey)
+		delete(labels, intreePvcNamespaceKey)
+		delete(labels, api.SpecParent)
 		cloneResp, err := volumes.Clone(ctx, &api.SdkVolumeCloneRequest{
 			Name:             req.GetName(),
 			ParentId:         source.Parent,
-			AdditionalLabels: clonedMetadata,
+			AdditionalLabels: labels,
```

Comment:
I am not entirely sure how the change is fixing the clone issue. Can you briefly describe what's happening?

Reply:
This is primarily so that other generic labels will be passed on - previously we were only going to pass the FADA pod value.

As discussed in slack, Porx PR coming soon which will help this make more sense.

Reply:
Yes changes look ok. Was just not able to understand how it helps. Will take a quick look at the Porx PR and approve this if I don't have any concerns.